lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -2145,6 +2166,59 @@ def construct_error_details_map_grpc(gax_error)
       {}
     end
 
+    # Construct monitored resource locally for k8s resources.
+    def construct_k8s_resource_locally(local_resource_id)
+      return unless
+        /^
+          (?<resource_type>k8s_container)
+          \.(?<namespace_name>[0-9a-z-]+)
+          \.(?<pod_name>[.0-9a-z-]+)
+          \.(?<container_name>[0-9a-z-]+)$/x =~ local_resource_id ||
+        /^
+          (?<resource_type>k8s_node)
+          \.(?<node_name>[0-9a-z-]+)$/x =~ local_resource_id
+
+      begin
+        @k8s_cluster_name ||= fetch_gce_metadata(
+          'instance/attributes/cluster-name')
+        @k8s_location ||= fetch_gce_metadata(
+          'instance/attributes/cluster-location')
+      rescue StandardError => e
+        @log.error 'Failed to retrieve k8s cluster name and location.', \
+                   error: e
+      end
+      case resource_type
+      when K8S_CONTAINER_CONSTANTS[:resource_type]
+        labels = {
+          'namespace_name' => namespace_name,
+          'pod_name' => pod_name,
+          'container_name' => container_name,
+          'cluster_name' => @k8s_cluster_name,
+          'location' => @k8s_location
+        }
+        fallback_resource = GKE_CONSTANTS[:resource_type]
+      when K8S_NODE_CONSTANTS[:resource_type]
+        labels = {
+          'node_name' => node_name,
+          'cluster_name' => @k8s_cluster_name,
+          'location' => @k8s_location
+        }
+        fallback_resource = COMPUTE_CONSTANTS[:resource_type]
+      end
+      unless @k8s_cluster_name && @k8s_location
+        @log.error "Failed to construct #{resource_type} resource locally." \
+                   ' Falling back to writing logs against' \
+                   " #{fallback_resource} resource.", error: e
+        return
+      end
+      constructed_resource = Google::Apis::LoggingV2::MonitoredResource.new(
+        type: resource_type,
+        labels: labels)
+      @log.debug("Constructed #{resource_type} resource locally:\n" \
```

Comment:
Keeping the log line on a single line (removing `\n`) would help debug later via pantheon log filters.

Reply:
+1, though we don't ingest our logs today, so I'm not sure exactly when this would help.

Reply:
Done.